conformance_tests/sysman/test_sysman_device/src/test_sysman_device_helper.cpp

```diff
@@ -74,22 +133,24 @@ int main(int argc, char **argv) {
   auto sysman_devices = lzt::get_zes_devices();
   EXPECT_FALSE(sysman_devices.empty());
 
-  std::vector<zes_uuid_t> sysman_device_uuids;
-  get_sysman_devices_uuids(sysman_devices, sysman_device_uuids);
+  if (strcmp(device_hierarchy, "FLAT") != 0) { // composite or combined mode
+    std::vector<zes_uuid_t> sysman_device_uuids;
+    get_sysman_devices_uuids(sysman_devices, sysman_device_uuids);
 
-  std::vector<ze_device_uuid_t> ze_root_uuids;
-  get_ze_root_uuids(ze_devices, ze_root_uuids, device_hierarchy);
+    std::vector<ze_device_uuid_t> ze_root_uuids;
+    get_ze_root_uuids(ze_devices, ze_root_uuids, device_hierarchy);
 
-  for (const auto &ze_root_uuid : ze_root_uuids) {
-    bool ze_and_sysman_uuid_equal = false;
-    for (const auto &sysman_device_uuid : sysman_device_uuids) {
-      if (memcmp(ze_root_uuid.id, sysman_device_uuid.id,
-                 ZE_MAX_DEVICE_UUID_SIZE) == false) {
-        ze_and_sysman_uuid_equal = true;
-        break;
-      }
-    }
-    EXPECT_TRUE(ze_and_sysman_uuid_equal);
+    compare_core_and_sysman_uuid(ze_root_uuids, sysman_device_uuids);
+  } else { // flat mode
+    std::vector<zes_uuid_t> sysman_sub_device_uuids;
+    // if subdevice doesn't exist for a device, then root device UUID
```

Comment:
This comment is already inside the function right ?
May be not necessary to repeat

Reply:
Ok will remove it